.circleci/config.yml

```diff
@@ -279,6 +279,20 @@ shared: &shared
     - run:
         name: Analyze Test Durations
         command: node misc/test-durations/ci
+    # Generate ESLint report
+    - run:
+        name: Run ESLint and generate ESLint report
+        command: |
+          if [ -n "$NODE_COVERAGE" ]; then
+            npx eslint packages/ -f json > eslint-report.json || exit 0
```

Comment:
Thank you for bringing this up. When it comes to SonarQube, it requires the ESLint report to be in JSON format, which is why I included the -f json flag. The existing step, on the other hand, performs additional tasks such as running TypeScript (tsc) and using ESLint via lerna. These additional tasks aren't necessary for SonarQube, and the existing step is common to other jobs as well. Therefore, I thought it might be a good idea to isolate this command exclusively for SonarQube. I'm open to alternative suggestions, so please feel free to share your thoughts as well.

https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/importing-external-issues/importing-third-party-issues/#:~:text=sonar.eslint.reportPaths